### PR TITLE
Add more information about mail errors

### PR DESCRIPTION
**Pull request by mre**
See #389

### CODE REVIEW

File: lychee-lib/src/types/mail.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        return format!("{:?}", e);
+    } else if let Err(e) = &o.mx {
+        return format!("{:?}", e);
+    }
+    match &o.is_reachable {
+            Reachable::Safe => "Safe: The email is safe to send",
+            Reachable::Risky => "Risky: The email address appears to exist, but has quality issues that may result in low engagement or a bounce",
+            Reachable::Invalid => "Invalid: Email doesn't exist or is syntactically incorrect",
+            Reachable::Unknown => "Unknown: We're unable to get a valid response from the recipient's email server."
+        }.to_string()
```
Comment (lebensterben) [inline lychee-lib/src/types/mail.rs:18]: doesn't seem like indented correctly

Comment (mre) [reply lychee-lib/src/types/mail.rs:18]: that's what cargo fmt gave me 🤷‍♀️

Comment (mre) [reply lychee-lib/src/types/mail.rs:18]: It's weird, though: when I change the indentation of the match blog, it doesn't change it. :/

Comment (lebensterben) [reply lychee-lib/src/types/mail.rs:18]: you might accidentally find a bug...

Comment (mre) [reply lychee-lib/src/types/mail.rs:18]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=242e40ce131f604a1b1334b86cbec615

Comment (lebensterben) [reply lychee-lib/src/types/mail.rs:18]: I know rustfmt gives up on really long lines. But I don't know it gives up so easily...

Comment (mre) [reply lychee-lib/src/types/mail.rs:18]: Here's a minimal code example:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=80660b19853fe958557723f5e00682dd

```
fn main() {
    match () {
    _ => "foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo foo",
};
}
```

Could be a rustfmt bug.

Comment (mre) [reply lychee-lib/src/types/mail.rs]: Manually changed the formatting to the way it _should_ look like IMHO.

**Pull request merged**